pkg/bundle/plain/v0/bundle.go

```diff
@@ -0,0 +1,382 @@
+package bundle
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"io/fs"
+	"strings"
+
+	"github.com/davecgh/go-spew/spew"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/rukpak/pkg/bundle"
+	registryv1 "github.com/operator-framework/rukpak/pkg/bundle/registry/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Bundle holds the contents of a plain+v0 bundle.
+type Bundle struct {
+	bundle.FS
+
+	// external settings
+	installNamespace string
+	targetNamespaces []string
```

Comment:
I got a little confused when I first started reading this package as the concept of namespacing, install modes and csv deployments are all `registry+v1` specific. I understand that all the functions relating to these concepts like `validateOptions()` and `supportedInstallModes()` are used to convert from a `registry+v1` bundle in `FromRegistryV1()`. However, I believe we can make this code a bit more readable for future contributors. I was thinking we could do one of three things:

1. Improve the comments around these functions describing why they are necessary and emphasizing that they are not a core part of what a `plain` bundle is (and are just for conversions)
2. In this `plain/v0` package add a new `bundle_conversion.go` file that contains these functions like normal in order to logically separate the concepts.
3. Move the conversion burden onto the `registry/v1` package and have something along the lines of `registryv1.ToPlainV0()` vs `plainv0.FromRegistryV1`.

Reply:
Thanks for the suggestions. One of my major concerns, probably _the_ major concern, is designing a really simple and understandable API so this is really helpful.

Reply:
Marking this as resolved. The comment + new way of converting Bundles feels very extensible and intuitive. Great work!